node/lib/util/add_submodule.js

```diff
@@ -65,20 +65,15 @@ exports.addSubmodule = co.wrap(function *(repo, url, filename, importArg) {
     }
     const modulesPath = path.join(repo.workdir(),
                                   SubmoduleConfigUtil.modulesFileName);
-    fs.appendFileSync(modulesPath, `\
-[submodule "${filename}"]
-    path = ${filename}
-    url = ${url}
-`);
     const index = yield repo.index();
-    yield index.addByPath(SubmoduleConfigUtil.modulesFileName);
-    yield index.write();
-
-    // sort all the urls since new modules get added
     const urls = yield SubmoduleConfigUtil.getSubmodulesFromIndex(repo, index);
+    urls[filename] = url;
     const newConf = SubmoduleConfigUtil.writeConfigText(urls);
```

Comment:
I ran this and it seems to remove the newline at the end of .gitmodules.

Reply:
There should be a local patch to perform our site-local sorting logic.